docs/source/workflows/data_sources.rst

```diff
@@ -15,15 +15,21 @@ Uploading a new file with the same name will produce a new version of the file w
 The columns in the CSV are extracted and parsed by a mapping of column header names to user-created descriptors.
 Columns in the CSV that are not mapped with a descriptor are ignored.
 
+An optional list of identifiers can be specified.
+Each identifier is a column header name.
+These may overlap with the keys defined in the mapping from column header names to descriptors if a column should be parsed as data and used as an identifier.
+Identifiers should be globally unique.
+No two rows should contain the same value.
```

Comment:
Is "should" strong enough here? If so, it would be helpful to spell out circumstances when it might be reasonable not to have unique rows.

Reply:
No, I'll clarify the wording in this and the previous sentence.